app/models/study.rb

```diff
@@ -1635,6 +1622,18 @@ def self.delete_all_and_remove_workspaces
     end
   end
 
+  # helper method that mimics DeleteQueueJob.delete_convention_data
+  # referenced from ensure_cascade_on_associations to prevent orphaned rows in BQ on manual deletes
```

Comment:
I like this method a lot, should we update DeleteQueueJob to call this method instead of its own?

Reply:
Originally I thought "no", but on reflection maybe it should.  I'll check it out.